src/main/scala/org/broadinstitute/dsde/workbench/sam/api/UserRoutesV2.scala

```diff
@@ -189,7 +189,17 @@ trait UserRoutesV2 extends SamUserDirectives with SamRequestContextDirectives {
         for {
           allowances <- userService.getUserAllowances(samUser, samRequestContext)
           maybeAttributes <- userService.getUserAttributes(samUser.id, samRequestContext)
-          termsOfServiceDetails <- tosService.getTermsOfServiceDetailsForUser(samUser.id, samRequestContext)
+          termsOfServiceDetails <- tosService.getTermsOfServiceDetailsForUser(samUser.id, samRequestContext).recoverWith {
+            case e: WorkbenchExceptionWithErrorReport if e.errorReport.statusCode == Option(StatusCodes.NotFound) =>
```

Comment:
Could we make `getTermsOfServiceDetailsForUser` return an `Option`? I'd love to not rely on exceptions for expected control flow. We should make the ToS Routes translate a `None` to a 404.